src/main/java/com/mraof/minestuck/entity/ai/attack/GroundSlamGoal.java

```diff
@@ -0,0 +1,110 @@
+package com.mraof.minestuck.entity.ai.attack;
+
+import com.mraof.minestuck.entity.AttackingAnimatedEntity;
+import com.mraof.minestuck.entity.animation.PhasedMobAnimation;
+import com.mraof.minestuck.util.MSSoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.ai.goal.Goal;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+
+import java.util.List;
+
+/**
+ * Alternative attack for when the target is out of standard range, results in players being flung into the air if they were on the ground and not crouching.
+ */
+public class GroundSlamGoal<T extends AttackingAnimatedEntity> extends AnimatedAttackWhenInRangeGoal<T>
+{
+	public static final int GROUP_SLAM_COOLDOWN = 20; //how long it should take before another nearby mob can ground slam
+	
+	public GroundSlamGoal(T entity, PhasedMobAnimation animation, float minRange, float maxRange, int actionCooldown)
+	{
+		super(entity, animation, minRange, maxRange, actionCooldown);
+	}
+	
+	@Override
+	public boolean canUse()
+	{
+		return super.canUse() && entity.hasFinishedCooldown();
+	}
+	
+	@Override
+	public void start()
+	{
+		super.start();
+		
+		Level level = this.entity.getLevel();
+		AABB aabb = new AABB(this.entity.blockPosition()).inflate(4);
+		List<AttackingAnimatedEntity> entityList = level.getEntitiesOfClass(AttackingAnimatedEntity.class, aabb);
+		if(!entityList.isEmpty())
+		{
+			for(AttackingAnimatedEntity iteratedEntity : entityList)
+			{
+				if(iteratedEntity != this.entity && !iteratedEntity.existingCooldownIsLonger(GROUP_SLAM_COOLDOWN))
+				{
+					//TODO slamGoal bool does not seem to trigger
+					for(Goal iteratedGoal : iteratedEntity.goalSelector.getAvailableGoals())
+					{
+						boolean slamGoal = iteratedGoal.equals(this);
+						if(slamGoal)
+							iteratedEntity.setCooldown(GROUP_SLAM_COOLDOWN + phasedAnimation.getTotalAnimationLength()); //plays at beginning to prevent overlapping
+					}
```

Comment:
Trying to make it so that it checks for all nearby entities, and if those nearby entities possess the GroundSlamGoal then a cooldown is also applied to them. Not sure if this is a bad approach or if I messed up the conditions of it since it doesn't seem to work

Reply:
No longer trying to give a cooldown via Goal. Now it just gives a group cooldown to entities of the same type

Reply:
Yeah, that sounds sensible for now.